src/Raven.Server/Documents/TimeSeries/TimeSeriesRollups.cs

```diff
@@ -766,9 +768,25 @@ void AggregateIndividualItems(IEnumerable<TimeSeriesStorage.Reader.SingleResult>
         }
         public static long NextRollup(DateTime time, TimeSeriesPolicy nextPolicy)
         {
-            var integerPart = time.Ticks / nextPolicy.AggregationTime.Ticks;
-            var nextRollup = nextPolicy.AggregationTime.Ticks * (integerPart + 1);
-            return nextRollup;
+            if (time == DateTime.MinValue)
+                return time.Add(nextPolicy.AggregationTime).Ticks;
+
+            if (nextPolicy.AggregationTime.Seconds != 0)
+            {
+                // align by seconds
+                var timespan = TimeSpan.FromSeconds(nextPolicy.AggregationTime.Seconds);
+                var integerPart = time.Ticks / timespan.Ticks;
+                var nextRollup = timespan.Ticks * (integerPart + 1);
+                return nextRollup;
+            }
+
+            // align by months
+            var totalMonths = time.Year * 12 + time.Month - 1;
+            var integerAggPart = totalMonths / nextPolicy.AggregationTime.Months;
+            var nextInMonths = nextPolicy.AggregationTime.Months * (integerAggPart + 1);
+            var years = nextInMonths / 12;
+            var months = nextInMonths % 12;
+            return new DateTime(years, months + 1, 1).Ticks;
```

Comment:
I wonder if we should also take into account the lap years?

Reply:
or actually we do, as years contains values around 2k, not 1-2 right? We add to current date?

Reply:
Months and days are isolated in this solution, you can't convert from days to months and via versa.

So we don't care about lap years here, any year in the gregorian calendar has 12 months.
Unlike the hebrew calendar that sometimes has 13 months :-)

Reply:
oh that make sense - thanks ;-)